inst/resources/activate.R

```diff
@@ -69,6 +69,25 @@ local({
 
   })
 
+  check_project_synchronization <- function() {
+
+    capture.output({status <- renv::status()})
+    if ("synchronized" %in% names(status) && !status$synchronized) {
+      msg <- "The project and lockfile are out of sync. Use `renv::status()` to view details."
+      warning(msg, call. = FALSE)
+    }
+
+  }
+
+  load_project <- function() {
```

Comment:
This is the wrong place for this check. Instead there should likely be something like `renv_load_check()` or similar here:

https://github.com/rstudio/renv/blob/18d1169f04370d37021fce7dbc188bd256ecde22/R/load.R#L361

which calls the relevant function to check project synchronization.

Reply:
Will do - I was under the impression we would only want a warning on session start and not subsequent loads.